libyul/AsmParser.cpp

```diff
@@ -249,6 +276,8 @@ ForLoop Parser::parseForLoop()
 	forLoop.pre = parseBlock();
 	forLoop.condition = make_unique<Expression>(parseExpression());
 	forLoop.post = parseBlock();
+	bool outsideForLoopBody = exchange(m_insideForLoopBody, true);
+	ScopeGuard forLoopBodyGuard{[=]() mutable { m_insideForLoopBody = outsideForLoopBody; }};
```

Comment:
This is unnecessary.

Reply:
setting the value back to what it was, I think is necessary. as it could be a nested loop.
```
for { } condition { }
{
    for {} condition {}
    {
        // ...
    }
    // here using break should be possible again
}
// here not
```

So for my understanding, I have to restore this state var to whatever it was before. Am I missing something?

Reply:
Sorry, I meant using the ScopeGuard to do that is unnecessary.